site/app/models/gradeable/GradeableList.php

```diff
@@ -251,4 +252,34 @@ public function getGradeablesBySection(): array {
             self::GRADED => $this->getGradedGradeables()
         ];
     }
+
+    /**
+     * A static factory method to create a new GradeableList object that contains
+     * all gradeables in all courses of a single user.
+     * The method loads from the database of all courses and get all gradeables information.
+     * Only load once unless the user refreshes the page.
+     *
+     * @param Core $core
+     * @param User $user The user to filter gradeables by
+     * @return GradeableList
+     * @throws \Exception if a Gradeable failed to load from the database
+     */
+    public static function getAllGradeableListFromUserId(Core $core, User $user): GradeableList {
+        $gradeables = [];
+        // Load the gradeable information for each course
+        $courses = $core->getQueries()->getCourseForUserId($user->getId());
+        foreach ($courses as $course) {
+            /** @var \app\models\Course $course */
+            $core->loadCourseConfig($course->getSemester(), $course->getTitle());
+            $core->loadCourseDatabase();
+            foreach ($core->getQueries()->getGradeableConfigs(null) as $gradeable) {
+                /** @var Gradeable $gradeable */
+                $gradeables["{$course->getSemester()}||{$course->getTitle()}||{$gradeable->getId()}"] = $gradeable;
```

Comment:
Doing it this way with the `||` means that the code will currently break if the course title has a `||` in it as it'll affect the `explode` you do elsewhere. Could we use the course code instead which is better guaranteed to not have a `|` character in it?